eng/common/scripts/modules/Package-Properties.psm1

```diff
@@ -0,0 +1,237 @@
+# Helper functions for retireving useful information from azure-sdk-for-* repo
+# Example Use : Import-Module .\eng\common\scripts\modules
+class PackageProps
+{
+    [string]$pkgName
+    [AzureEngSemanticVersion]$pkgVersion
+    [string]$pkgDirectoryPath
+    [string]$pkgReadMePath
+    [string]$pkgChangeLogPath
+
+    PackageProps(
+        [string]$pkgName,
+        [string]$pkgVersion,
+        [string]$pkgDirectoryPath
+    )
+    {
+        $this.pkgName = $pkgName
+        $this.pkgVersion = [AzureEngSemanticVersion]::new($pkgVersion)
+        $this.pkgDirectoryPath = $pkgDirectoryPath
+
+        if (Test-Path (Join-Path $pkgDirectoryPath "README.md"))
+        {
+            $this.pkgReadMePath = Join-Path $pkgDirectoryPath "README.md"
+        } 
+        else
+        {
+            $this.pkgReadMePath = $null
+        }
+
+        if (Test-Path (Join-Path $pkgDirectoryPath "CHANGELOG.md"))
+        {
+            $this.pkgChangeLogPath = Join-Path $pkgDirectoryPath "CHANGELOG.md"
+        } 
+        else
+        {
+            $this.pkgChangeLogPath = $null
+        }
+    }
+}
+
+Install-Module -Name powershell-yaml -RequiredVersion 0.4.1 -Force -Scope CurrentUser
+
+function Extract-PkgProps ($pkgPath, $pkgName)
+{
+    if ($Language -eq "net")
```

Comment:
We should probably pass language instead of depend on the global value. 

As a potential alternative approach see what I did in https://github.com/Azure/azure-sdk/blob/master/eng/scripts/Update-Release-Versions.ps1#L302, you can essentially build the function name in a string and then call it.

Reply:
Fixed